project/SparkBuild.scala

```diff
@@ -413,6 +413,9 @@ object SparkBuild extends PomBuild {
 
   enable(KubernetesIntegrationTests.settings)(kubernetesIntegrationTests)
 
+  if (profiles.contains("hadoop-provided")) {
+    sys.props("spark.yarn.isHadoopProvided") = "true"
+  }
```

Comment:
nit but would it be better to move this logic into the `YARN`?

Reply:
`profiles` cannot be referred from `YARN` but I found another better way.